src/datepicker.ts

```diff
@@ -632,6 +631,9 @@ export class Datepicker extends LitElement {
         if (focusedDateTime > maxTime) newValue = maxTime;
 
         this.value = toFormattedDateString(new Date(newValue));
+      } else if (getDateRange(minTime, maxTime) === 0) {
```

Comment:
I think this is just when both `min` and `max` is the same day regardless of the time so this should work.

```ts
// This means `min` and `max` is the same day.
} else {
   ...
}
```